### PR TITLE
Add online_rescue_image

### LINKED ISSUE #5: Make rescue image data provider
We should add a data provider which is able to select the correct rescue image based on the name of the rescue image.

The images are different per server and can be obtained by calling ` /server/rescue_images/{server_id}`

### PR DESCRIPTION
**Pull request by meyskens**
Fixes #5

### CODE REVIEW

File: provider/data_rescue_image.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		return err
+	}
+
+	selectedImage := ""
+
+	if hasName {
+		want := nameInterface.(string)
+		for _, image := range images {
+			if image == want {
+				selectedImage = image
```
Comment (rporres) [inline provider/data_rescue_image.go]: you could break here

File: provider/data_rescue_image.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+			}
+		}
+	}
+
+	if hasNameFilter {
+		want := nameFilterInterface.(string)
+		for _, image := range images {
+			if strings.Contains(image, want) {
+				selectedImage = image
```
Comment (rporres) [inline provider/data_rescue_image.go]: I understand you cannot break here as we may want the last matching image, but at least this behaviour should be documented somewhere

File: provider/data_rescue_image.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		return errors.New("Need either a name or a name_filter")
+	}
+
+	images, err := c.GetRescueImages(server)
+	if err != nil {
+		return err
+	}
+
+	selectedImage := ""
+
```
Comment (rporres) [inline provider/data_rescue_image.go]: I would do if/else if here

File: provider/data_rescue_image.go (hunk 2 above)
Comment (meyskens) [reply provider/data_rescue_image.go]: Added in description

**Pull request merged**